addons/mail/static/src/models/channel_member.js

```diff
@@ -30,7 +30,7 @@ registerModel({
                 return !this.persona.partner.isOnline ? replace(this.channel) : clear();
             }
             if (this.persona.guest) {
-                return replace(this.channel);
+                return !this.persona.guest.isOnline ? replace(this.channel) : clear();
```

Comment:
this is confusing by using a ? conditional line but starts with a not sentence like `!this.persona.guest.isOnline`. it will be easier to read and understand if reverse the conditions and deleted the `not` operator.

Reply:
It follow the same pattern has the partner condition below. I don't think we need to overthink those simple condition.